MarkBind/markbind
# Fix image linkify and alt attribute

**What is the purpose of this pull request?**




**Overview of changes:**
Fixes #2045 
Fixes #2028

Addressing two related issues on images
- linkify
  - document the linkify feature for the image syntax via `![]()` 
  - fixed the upstream repo to deal with the double wrapping issue https://github.com/crookedneighbor/markdown-it-linkify-images/pull/46
- alt not captured
  - seems like due to outdated code, markdown-it-imsize does not correctly set the content of the image such that it can be passed into the "alt" attribute when rendered as HTML
  - I forked and published a patched version of it since the package has quite a bit of code and is not very convenient to patch within markbind (but can be done in the future if there's such a need)

**Anything you'd like to highlight/discuss:**
The default markdown image syntax seems to overlap with the features of the `pic` component, and some features exist on one and not the other. I think it is still important to support the default syntax since it's a very common usage. Further improvements can be done to the image support in the future.

**Testing instructions:**
- Verify added functional test in packages/cli/test/functional/test_site/testImages.md
- Verify updated docs: https://deploy-preview-2046--markbind-master.netlify.app/userguide/formattingcontents#images



**Proposed commit message: (wrap lines at 72 characters)**
Fix image linkify and alt attribute

Currently the image linkify feature is not documented and does
not account for already wrapped images. The image alternative
text is also not captured due to outdated markdown-it-imsize.

Let's document the linkify feature and update the relevant
packages to fix the issues. Let's also add tests to ensure the
functionalities are captured and verified.



---

**Checklist:** :ballot_box_with_check:
